rh-summit-2018/module-03.adoc

```diff
@@ -1,3 +1,296 @@
+        "connection.url": "jdbc:postgresql://postgresql-95-centos7:5432/inventory?user=postgresuser&password=postgrespw",
+        "transforms": "unwrap",
+        "transforms.unwrap.type": "io.debezium.transforms.UnwrapFromEnvelope",
+        "auto.create": "true",
+        "insert.mode": "upsert",
+        "pk.fields": "id",
+        "pk.mode": "record_value"
+    }
+}
+EOF
+----
+
+This sets up an an instance of `io.confluent.connect.jdbc.JdbcSinkConnector`,
+listening to the `dbserver1.inventory.Hike` and streaming all data changes to the given database connection.
+As this sink connector just expects the effective state of changed rows
+(i.e. the "after" part from the Debezium data change messages),
+only this part is extracted using Debezium's `UnwrapFromEnvelope` SMT (single message transform).
+
+With the sink connector being set up, we can take a look into the Postgres database and see how the table changes are propgated there.
+Get a shell on the pod of the Postgres service:
+
+[source]
+----
+oc rsh $(oc get pods -o name -l app=postgresql-95-centos7)
+----
+
+Run a query to get all records from the table corresponding to the monitored topic:
+
+[source]
+----
+psql -U postgresuser inventory -c 'select * from "dbserver1.inventory.Hike"'
+----
+
+As you alter records in the source web application,
+you'll see how the table in Postgres gets updated accordingly, if you re-execute the query.
+Note that `DELETE` operations currently cannot be propagated, as they are not yet supported by the Confluent JDBC sink connector.
+
+To leave the shell on the Postgres pod, run:
+
+[source]
+----
+exit
+----
+
+As an experiment, you also can explore how the streaming approach ensures a loose coupling of the involved components.
+Scale down the pods of the "strimzi-connect-s2i" application to 0:
+
+[source]
+----
+oc scale --replicas=0 dc/debezium-connect
+----
+
+You'll still be able to edit records in the source application,
+but as Kafka Connect - and with it Debezium - isn't running,
+the changes won't be propagated to the sink database.
+
+Once Kafka Connect is restarted again, the connector will automatically pick up where it left before and after a while,
+you'll see all changes that had occurred in the connector's downtime in the sink database:
+
+[source]
+----
+oc scale --replicas=1 dc/debezium-connect
+----
+
+=== Consuming Data Change Events With WildFly Swarm
+
+Finally, let's explore how to consume the Debezium events in a custom application and forward them to a web UI using WebSockets.
+
+Deploy the application:
+
+[source]
+----
+oc new-app --name=websocketsinkapp fabric8/s2i-java:latest~https://github.com/strimzi/strimzi-lab.git \
+    --context-dir=debezium-swarm-demo \
+    -e MYSQL_DATABASE=inventory \
+    -e AB_PROMETHEUS_OFF=true \
+    -e KAFKA_SERVICE_HOST=broker-kafka \
+    -e KAFKA_SERVICE_PORT=9092
+
+oc expose svc websocketsinkapp
+----
+
+One more more must be exposed for the application.
```

Comment:
@jpechane, do you happen to know what'd be the equivalent on the CLI (`oc`) for the following two steps, which I'm describing in the OpenShift web console here?

Reply:
`oc expose` can create both services and routes